src/hooks/useOrders.ts

```diff
@@ -50,13 +50,24 @@ export function useOrders(): Result {
   //  consider first state to be loading
   const [isLoading, setIsLoading] = useSafeState<boolean>(true)
 
+  useEffect(() => {
+    // continue loading new orders
+    // from current offset
+    setIsLoading(true)
+    // whenever new block is mined
+  }, [blockNumber, setIsLoading])
+
   useEffect(() => {
     let cancelled = false
 
     const fetchOrders = async (offset: number): Promise<void> => {
       // isLoading is the important one
       // controls ongoing fetching chain
-      if (!userAddress || !networkId || !isLoading) return
+      if (!userAddress || !networkId || !isLoading) {
+        // next isLoading = true will be when userAddress and networkId are valid
+        setIsLoading(false)
```

Comment:
I solved this a little differently https://github.com/gnosis/dex-react/commit/30ae106415f125232afbff3d915f277f99876f53
Regardless, to avoid a re-render, can you make this state change conditional?
Or does React take care of that already (by not re-rendering when the state is set to the same value)?

Reply:
yes,` setState(oldValue)` is noop

Reply:
Also for the complete fix this `setIsLoading` should come after the other one in `useEffect` 34b33cd